### ISSUE TITLE
Any complaints against adding nimgit and the libgit2 dependencies?

### ISSUE THREAD

**Issue by MisterBianco** [icyphox/nicy#11]
I have been using nicy for a bit with some modifications particularly around the gitBranch and gitStatus procs. I have one proc that gets both as a tuple and uses libgit2.

This is a beefy dependency and really changes nicy, if you are intereseted I can submit the code and let you decide what to do with it.

Time tests:
---

```nim
let t0 = getTime()
for x in countup(0, 1000):
  discard gitInfo()
echo getTime() - t0

let t1 = getTime()
for x in countup(0, 1000):
  discard gitBranch()
  discard gitStatus("*", "")
echo getTime() - t1
```

Times show:
---

```
66 milliseconds and 781 microseconds
13 seconds, 485 milliseconds, and 359 microseconds
```

Speed isnt everything, but those are the stats.

**Comment by icyphox** [comment]
> This is a beefy dependency and really changes nicy, if you are
> intereseted I can submit the code and let you decide what to do with it.

I'm not sure really. It does sound better than shelling out though. As
long as it doesn't break existing users' prompts -- I'm cool with it. I
don't use nicy myself anymore; I think I'll loop in @juancarlospaco just
for a second opinion.

**Comment by juancarlospaco** [comment]
```nim
when defined(nimgit):
  # nimgit
else:
  # old code
```

:question:

**Comment by icyphox** [comment]
Good idea. That could work. @MisterBianco, do you think you can do that?

**Comment by juancarlospaco** [comment]
BTW What are you using Fox?, I try Xonsh too, but Python is too slow, too much input lag.

**Comment by icyphox** [comment]
I'm using plain bash right now. It just works, solves all my needs.
--
Anirudh Oppiliappan
https://icyphox.sh
